packages/dds/tree/src/test/domains/json/benchmarks.ts

```diff
@@ -28,9 +24,7 @@ export function sumMap(cursor: ITreeCursorNew): number {
         total += value;
     }
 
-    for (let moreFields = cursor.firstField(); moreFields; moreFields = cursor.nextField()) {
-        forEachNode(cursor, sumMap);
-    }
+    forEachField(cursor, () => forEachNode(cursor, sumMap));
```

Comment:
how does this even work?

The point of having sum and sum map was to have one that looped directly, and one that used the higher order functions so we could compare their performance. This change makes both use the higher order functions, which removes the reason for having both. Looks like we should add some comments clarify what they are for, and maybe rename them.

Looks like sum map was already broken and didn't actually compute the sum properly: we should probably  have an assert or test that it actually works somewhere. This change does not appear to fix it.

Reply:
Ah, I wasn't familiar with the intention of the sum benchmarks. I can definitely change it back and just make it correct. The higher order function and the correct version of the previous code seem to be basically the same though so is it worth benchmarking just to see if an additional function call is slow? I notice the legacy version doesn't even have a sumMap.

Reply:
The old API made it impractical to work without the helper functions, so it wasn't worth comparing the two options.

I think here what we want is:

```suggestion
    forEachField(cursor, () => forEachNode(cursor, (c) => { total += sumMap(c); }));
```

Thats a significant amount of extra functions and closures (I think a 4x deeper call stack for example), so it could have some cost. In the past I measured it to be significant, but small. Thus I preferred to use the for loop syntax where it mattered over these helpers.